pool/app/guard/repo.ml

```diff
@@ -113,6 +114,81 @@ module RolePermission = struct
   ;;
 
   let insert pool = insert ~ctx:(Pool_database.to_ctx pool)
+
+  let find_by_target_and_permissions_request permissions =
+    let select_from_actor_roles =
+      {sql|
+        SELECT
+          guardian_actors.uuid AS uuid,
+          guardian_role_permissions.target_model AS target_model,
+          guardian_role_permissions.permission AS permission,
+          guardian_actor_roles.mark_as_deleted AS mark_as_deleted
+        FROM
+          guardian_actors
+          INNER JOIN guardian_actor_roles ON guardian_actor_roles.actor_uuid = guardian_actors.uuid
+          INNER JOIN guardian_role_permissions ON guardian_role_permissions.role = guardian_actor_roles.role
+      |sql}
```

Comment:
Within guardian, I already exclude "mark as deleted" entities when joining the tables.

```suggestion
      {sql|
        SELECT
          guardian_actors.uuid AS uuid,
          guardian_role_permissions.target_model AS target_model,
          guardian_role_permissions.permission AS permission,
        FROM
          guardian_actors
          INNER JOIN guardian_actor_roles ON guardian_actor_roles.mark_as_deleted IS NULL AND guardian_actor_roles.actor_uuid = guardian_actors.uuid
          INNER JOIN guardian_role_permissions ON guardian_role_permissions.mark_as_deleted IS NULL AND guardian_role_permissions.role = guardian_actor_roles.role
      |sql}
```